tests/e2e/gc_mgmt_wrkld_domain_isolation.go

```diff
@@ -0,0 +1,351 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/kubernetes/test/e2e/framework"
+	fss "k8s.io/kubernetes/test/e2e/framework/statefulset"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	snapclient "github.com/kubernetes-csi/external-snapshotter/client/v8/clientset/versioned"
+)
+
+var _ bool = ginkgo.Describe("[domain-isolation] Management-Workload-Domain-Isolation", func() {
+
+	f := framework.NewDefaultFramework("domain-isolation")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+	var (
+		client                  clientset.Interface
+		namespace               string
+		vcRestSessionId         string
+		allowedTopologies       []v1.TopologySelectorLabelRequirement
+		replicas                int32
+		topologyAffinityDetails map[string][]string
+		topologyCategories      []string
+		labelsMap               map[string]string
+		labels_ns               map[string]string
+		pandoraSyncWaitTime     int
+		snapc                   *snapclient.Clientset
+		err                     error
+		zone2                   string
+		zone3                   string
+		sharedStoragePolicyName string
+		gcNamespace             string
+		guestClusterRestConfig  *restclient.Config
+	)
+
+	ginkgo.BeforeEach(func() {
+		namespace = getNamespaceToRunTests(f)
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		// making vc connection
+		client = f.ClientSet
+		bootstrap()
+
+		// reading vc session id
+		if vcRestSessionId == "" {
+			vcRestSessionId = createVcSession4RestApis(ctx)
+		}
+
+		// reading topology map set for management doamin and workload domain
+		topologyMap := GetAndExpectStringEnvVar(envTopologyMap)
+		allowedTopologies = createAllowedTopolgies(topologyMap)
+		topologyAffinityDetails, topologyCategories = createTopologyMapLevel5(topologyMap)
+
+		// required for pod creation
+		labels_ns = map[string]string{}
+		labels_ns[admissionapi.EnforceLevelLabel] = string(admissionapi.LevelPrivileged)
+		labels_ns["e2e-framework"] = f.BaseName
+
+		//setting map values
+		labelsMap = make(map[string]string)
+		labelsMap["app"] = "test"
+
+		// reading fullsync wait time
+		if os.Getenv(envPandoraSyncWaitTime) != "" {
+			pandoraSyncWaitTime, err = strconv.Atoi(os.Getenv(envPandoraSyncWaitTime))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		} else {
+			pandoraSyncWaitTime = defaultPandoraSyncWaitTime
+		}
+
+		//zones used in the test
+		zone2 = topologyAffinityDetails[topologyCategories[0]][1]
+		zone3 = topologyAffinityDetails[topologyCategories[0]][2]
+
+		// reading shared storage policy
+		sharedStoragePolicyName = GetAndExpectStringEnvVar(envIsolationSharedStoragePolicyName)
+		if sharedStoragePolicyName == "" {
+			ginkgo.Skip("Skipping the test because WORKLOAD_ISOLATION_SHARED_STORAGE_POLICY is not set")
+		}
+
+		gcNamespace = GetAndExpectStringEnvVar(envSupervisorClusterNamespace)
```

Comment:
Let the name be svcNamespace. gcNamespace is local to GC and is typically not exported; it gets created at runtime